api/src/main/java/org/openmrs/module/appointmentscheduling/api/impl/AppointmentServiceImpl.java

```diff
@@ -1218,4 +1301,41 @@ private Date setupDate(Date date, int hour, int minute, int second) {
 
 		return boundaries;
 	}
+
+	public List<AppointmentDailyCount> getAppointmentsDailyReport(List<Appointment> appointments) throws ParseException {
+		List<AppointmentDailyCount> appointmentDailyCountList = new ArrayList<AppointmentDailyCount>();
+		ListIterator<Appointment> iterator = appointments.listIterator();
+		ArrayList<Date> dates = new ArrayList<Date>();
+		while (iterator.hasNext()) {
+			dates.add(new SimpleDateFormat("yyyy-MM-dd").parse(new SimpleDateFormat("yyyy-MM-dd").format(iterator.next().getTimeSlot().getStartDate())));//2019-05-07
```

Comment:
What is the purpose of this line - formatting and immediately parsing?

Reply:
the timeslots dates contains both the date and time, I only need the dates without times so that I could do the comparisons that's why i'm formatting and passing